source/Calamari.Tests/Fixtures/PackageRetention/JsonJournalRepositoryFixture.cs

```diff
@@ -0,0 +1,63 @@
+﻿using System;
+using System.IO;
+using Calamari.Common.Features.Processes.Semaphores;
+using Calamari.Common.Plumbing.Variables;
+using Calamari.Deployment.PackageRetention.Repositories;
+using Calamari.Tests.Fixtures.Integration.FileSystem;
+using Calamari.Tests.Helpers;
+using NSubstitute;
+using NUnit.Framework;
+
+namespace Calamari.Tests.Fixtures.PackageRetention
+{
+    [TestFixture]
+    public class JsonJournalRepositoryFixture
+    {
+        static readonly string TentacleHome = TestEnvironment.GetTestPath("Fixtures", "JsonJournalRepository");
+
+        [SetUp]
+        public void SetUp()
+        {
+            if (!Directory.Exists(TentacleHome))
+                Directory.CreateDirectory(TentacleHome);
+        }
+
+        [TearDown]
+        public void TearDown()
+        {
+            if (Directory.Exists(TentacleHome))
+                Directory.Delete(TentacleHome, true);
+        }
+
+        [Test]
+        public void WhenCalamariPackageRetentionJournalPathExists_ThenTheJournalIsCreatedAtTheGivenPath()
```

Comment:
I think these tests should actually be revised. We don't really care _how_ the repository implements things, just that it does what it should.   So if we pass the variables like you have, then write something and read it back correctly, then we know that it works, without worrying about the implementation.

Reply:
That makes sense. I've updated the test to load, write something, then re-load the journal to check that what's there is what we expect. Also removed the `internal` from that default journal name since we no longer need it for the test.

Reply:
I think these are a lot better.   Given that these are almost completely the same code-wise, it could be worth using [test cases](https://docs.nunit.org/articles/nunit/writing-tests/attributes/testcase.html) for the differences, and specify the name with the TestName parameter on the test case attributes?

Reply:
Oh yeah true 😄 looks like a good use case for that.